.github/workflows/_shared-first-time-setup.yml

```
@@ -0,0 +1,65 @@
+name: "Repository First Time Setup automation (shared)"
+
+on:
+  workflow_call:
+    inputs:
+        esf_version:
+          type: string
+          default: '7.2.2'
+          required: true
+        archetype_url:
+          type: string
+          default: 'https://changeme'
+          required: true
+        groupId:
+          type: string
+          default: ''
+          required: true
+        artifactId:
+          type: string
+          default: ''
+          required: true
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Download ESF archetype
+      if: startsWith(${{ github.event.inputs.archetype_url }}, 'https://eth-repo.s3.amazonaws.com/esf')
+      uses: carlosperate/download-file-action@v1
+      id: download-file
+      with:
+        file-url: ${{ github.event.inputs.archetype_url }}
+
+    - name: Run the maven install
+      run: |
+        mvn install:install-file -Dfile=${{ steps.download-file.outputs.file-path }} \
+        -DgroupId=com.eurotech.framework -DartifactId=esf-addon-archetype \
+        -Dversion=${{ github.event.inputs.esf_version }} -Dpackaging=jar
+      shell: bash
+
+    - name: Run the maven archetype generate
+      run: |
+        mvn archetype:generate -B -DarchetypeArtifactId=esf-addon-archetype \
+        -DarchetypeGroupId=com.eurotech.framework -DarchetypeVersion="${{ github.event.inputs.esf_version }}" \
+        -DgroupId=${{ github.event.inputs.groupId }} -DartifactId=${{ github.event.inputs.artifactId }}
+      shell: bash
+
+    - name: Copy artifactId-named folder content to the main project directory
+      run: |
+        mv ${{ github.event.inputs.artifactId }}/* .
+      shell: bash
+
+    - name: Cleanup
+      run: |
+        rm ${{ github.event.inputs.artifactId }} ${{ steps.download-file.outputs.file-path }}
```

Since it's a directory we need something like:

```suggestion
        rm -rf ${{ github.event.inputs.artifactId }} ${{ steps.download-file.outputs.file-path }}
```